ui/pages/confirmations/components/confirm/header/header.tsx

```diff
@@ -17,12 +24,116 @@ import { getAvatarNetworkColor } from '../../../../../helpers/utils/accounts';
 import useConfirmationNetworkInfo from '../../../hooks/useConfirmationNetworkInfo';
 import useConfirmationRecipientInfo from '../../../hooks/useConfirmationRecipientInfo';
 import HeaderInfo from './header-info';
+import { useConfirmContext } from '../../../context/confirm';
+import {
+  TransactionMeta,
+  TransactionType,
+} from '@metamask/transaction-controller';
+import { Confirmation } from '../../../types/confirm';
+import { useDispatch, useSelector } from 'react-redux';
+import { editExistingTransaction } from '../../../../../ducks/send';
+import { AssetType } from '../../../../../../shared/constants/transaction';
+import { clearConfirmTransaction } from '../../../../../ducks/confirm-transaction/confirm-transaction.duck';
+import {
+  setConfirmationAdvancedDetailsOpen,
+  showSendTokenPage,
+} from '../../../../../store/actions';
+import { SEND_ROUTE } from '../../../../../helpers/constants/routes';
+import { useHistory } from 'react-router-dom';
+import { selectConfirmationAdvancedDetailsOpen } from '../../../selectors/preferences';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
 
 const Header = () => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const history = useHistory();
+
   const { networkImageUrl, networkDisplayName } = useConfirmationNetworkInfo();
   const { senderAddress: fromAddress, senderName: fromName } =
     useConfirmationRecipientInfo();
 
+  const { currentConfirmation } = useConfirmContext<Confirmation>();
+
+  const showAdvancedDetails = useSelector(
+    selectConfirmationAdvancedDetailsOpen,
+  );
+
+  const setShowAdvancedDetails = (value: boolean): void => {
+    dispatch(setConfirmationAdvancedDetailsOpen(value));
+  };
+
+  if (currentConfirmation.type === TransactionType.tokenMethodTransfer) {
+    const isWalletInitiated =
+      (currentConfirmation as TransactionMeta).origin === 'metamask';
+
+    if (isWalletInitiated) {
+      const onClickBackHandler = () => {
+        const handleEditTransaction = async ({
```

Comment:
As this is a handler rather than a `useEffect`, can the entire listener not be `async` and we just await this rather than defining a nested async function?